marklogic-unit-test-modules/src/main/ml-modules/root/test/test-controller.xqy

```diff
@@ -21,16 +21,27 @@ declare variable $root as xs:string := xdmp:modules-root();
  : Returns a list of the available tests. This list is magically computed based on the modules
  :)
 declare function list()
+{
+  list(fn:false())
+};
+(:
+ : Returns a list of the available tests. This list is magically computed based on the modules
+ :)
+declare function list($include-suite-setup-teardown as xs:boolean)
 {
 	let $directories-to-ignore := map:new((
     ".svn", "CVS", ".DS_Store", "Thumbs.db", "thumbs.db", "test-data", "lib"
   ) ! map:entry(., .))
 
-  let $files-to-ignore := map:new((
-		"setup.xqy", "teardown.xqy", "setup.sjs", "teardown.sjs",
-		"suite-setup.xqy", "suite-teardown.xqy", "suiteSetup.sjs", "suiteTeardown.sjs"
-	) ! map:entry(., .))
-
+  let $files-to-ignore-check := function ($file-name) {
+    fn:matches($file-name, "^(setup|teardown)\.")
```

Comment:
It looks like these patterns are used in several places. How about adding functions to test-helper.xqy named `is-setup-module($name)`, `is-teardown-module($name)`, `is-suite-setup-module($name)`, and `is-suite-teardown-module($name)`? That would avoid duplication of the patterns and make the code more readable.